pkg/backend/filestate/backend.go

```diff
@@ -321,7 +322,7 @@ func (b *localBackend) GetStack(ctx context.Context, stackRef backend.StackRefer
 	stackName := stackRef.Name()
 	snapshot, path, err := b.getStack(stackName)
 	switch {
-	case gcerrors.Code(errors.Cause(err)) == gcerrors.NotFound:
+	case gcerrors.Code(errors.Unwrap(err)) == gcerrors.NotFound:
```

Comment:
`errors.Unwrap` returns `nil` if the error isn't wrapped (i.e. doesn't have an `Unwrap` method):

https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/errors/wrap.go;l=14-22

Given the way `getStack` is implemented, I think we need to loop through calling `Unwrap` until it returns nil, and then return the previous err. Similar to what `errors.Is` and `errors.As` does:

https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/errors/wrap.go;l=14-22

Right?

Reply:
It looked to me like we wrapped any returned error, but I may have missed something. I can add that functionality.